From 4d5bca60322944e27d760dd95e109669f7c2bf13 Mon Sep 17 00:00:00 2001
From: Rui Salvaterra <rsalvaterra@gmail.com>
Date: Wed, 15 Sep 2021 10:31:37 +0100
Subject: [PATCH] kernel: bump 5.10 to 5.10.65

Manually refreshed patches:
generic/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch
platform/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch
platform/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>
---
 include/kernel-version.mk                     |  4 +--
 ...0015-firmware-Updated-mailbox-header.patch |  2 +-
 ...50-0043-fbdev-add-FBIOCOPYAREA-ioctl.patch |  8 +++---
 ...irmware-bcm2835-Support-ARCH_BCM270x.patch | 25 ++++++++++---------
 ...ware-Export-the-general-transaction-.patch |  4 +--
 ...pberrypi-Notify-firmware-of-a-reboot.patch |  6 ++---
 ...rypi-Add-backward-compatible-get_thr.patch | 12 ++++-----
 ...rypi-Report-the-fw-variant-during-pr.patch |  6 ++---
 ...lise-rpi-firmware-before-clk-bcm2835.patch |  2 +-
 ...-0201-mmc-sdhci-Silence-MMC-warnings.patch |  4 +--
 ...rypi-Add-support-for-tryonce-reboot-.patch |  4 +--
 ...the-full-state-to-CRTC-atomic-enable.patch |  4 +--
 ...-the-full-state-to-CRTC-atomic_check.patch |  2 +-
 ...the-full-state-to-CRTC-atomic-begin-.patch |  2 +-
 ...ate-pointer-directly-in-atomic_check.patch |  2 +-
 .../905-BCM53573-minor-hacks.patch            |  4 +--
 .../410-block-fit-partition-parser.patch      |  2 +-
 .../hack-5.10/904-debloat_dma_buf.patch       |  2 +-
 ...ng-with-source-address-failed-policy.patch | 18 ++++++-------
 ...ice-struct-copy-its-DMA-params-to-th.patch | 10 ++------
 ...r-Gateworks-PLX-PEX860x-switch-with-.patch |  4 +--
 21 files changed, 61 insertions(+), 66 deletions(-)

diff --git a/include/kernel-version.mk b/include/kernel-version.mk
index 00f7c90313b3..122cd8b0d6a5 100644
--- a/include/kernel-version.mk
+++ b/include/kernel-version.mk
@@ -7,10 +7,10 @@ ifdef CONFIG_TESTING_KERNEL
 endif
 
 LINUX_VERSION-5.4 = .145
-LINUX_VERSION-5.10 = .64
+LINUX_VERSION-5.10 = .65
 
 LINUX_KERNEL_HASH-5.4.145 = 5cf7782ec2e91417edf0d5e6555da6d556962c8985e33ba9e7dadba5cbdc68f9
-LINUX_KERNEL_HASH-5.10.64 = 3eb84bd24a2de2b4749314e34597c02401c5d6831b055ed5224adb405c35e30a
+LINUX_KERNEL_HASH-5.10.65 = edd3dedbce5bcaa5ba7cde62f8f3fd58b2ab21e2ec427b9d200685da5ec03e66
 
 remove_uri_prefix=$(subst git://,,$(subst http://,,$(subst https://,,$(1))))
 sanitize_uri=$(call qstrip,$(subst @,_,$(subst :,_,$(subst .,_,$(subst -,_,$(subst /,_,$(1)))))))
diff --git a/target/linux/bcm27xx/patches-5.10/950-0015-firmware-Updated-mailbox-header.patch b/target/linux/bcm27xx/patches-5.10/950-0015-firmware-Updated-mailbox-header.patch
index ec1e48a047b0..d909c9d150c1 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0015-firmware-Updated-mailbox-header.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0015-firmware-Updated-mailbox-header.patch
@@ -18,7 +18,7 @@ Subject: [PATCH] firmware: Updated mailbox header
  struct rpi_firmware;
  
  enum rpi_firmware_property_status {
-@@ -159,5 +161,6 @@ static inline struct rpi_firmware *rpi_f
+@@ -161,5 +163,6 @@ static inline struct rpi_firmware *rpi_f
  	return NULL;
  }
  #endif
diff --git a/target/linux/bcm27xx/patches-5.10/950-0043-fbdev-add-FBIOCOPYAREA-ioctl.patch b/target/linux/bcm27xx/patches-5.10/950-0043-fbdev-add-FBIOCOPYAREA-ioctl.patch
index 75e3aea2d75e..0807512c7246 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0043-fbdev-add-FBIOCOPYAREA-ioctl.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0043-fbdev-add-FBIOCOPYAREA-ioctl.patch
@@ -244,7 +244,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
  static int bcm2708_fb_register(struct bcm2708_fb *fb)
 --- a/drivers/video/fbdev/core/fbmem.c
 +++ b/drivers/video/fbdev/core/fbmem.c
-@@ -1079,6 +1079,30 @@ fb_blank(struct fb_info *info, int blank
+@@ -1085,6 +1085,30 @@ fb_blank(struct fb_info *info, int blank
  }
  EXPORT_SYMBOL(fb_blank);
  
@@ -275,7 +275,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
  static long do_fb_ioctl(struct fb_info *info, unsigned int cmd,
  			unsigned long arg)
  {
-@@ -1087,6 +1111,7 @@ static long do_fb_ioctl(struct fb_info *
+@@ -1093,6 +1117,7 @@ static long do_fb_ioctl(struct fb_info *
  	struct fb_fix_screeninfo fix;
  	struct fb_cmap cmap_from;
  	struct fb_cmap_user cmap;
@@ -283,7 +283,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
  	void __user *argp = (void __user *)arg;
  	long ret = 0;
  
-@@ -1162,6 +1187,15 @@ static long do_fb_ioctl(struct fb_info *
+@@ -1168,6 +1193,15 @@ static long do_fb_ioctl(struct fb_info *
  		unlock_fb_info(info);
  		console_unlock();
  		break;
@@ -299,7 +299,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
  	default:
  		lock_fb_info(info);
  		fb = info->fbops;
-@@ -1307,6 +1341,7 @@ static long fb_compat_ioctl(struct file
+@@ -1313,6 +1347,7 @@ static long fb_compat_ioctl(struct file
  	case FBIOPAN_DISPLAY:
  	case FBIOGET_CON2FBMAP:
  	case FBIOPUT_CON2FBMAP:
diff --git a/target/linux/bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch b/target/linux/bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch
index 08d1e2af2df5..b5b097209b61 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0053-firmware-bcm2835-Support-ARCH_BCM270x.patch
@@ -27,8 +27,8 @@ Signed-off-by: Noralf Trønnes <noralf@tronnes.org>
 
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -29,6 +29,8 @@ struct rpi_firmware {
- 	u32 enabled;
+@@ -32,6 +32,8 @@ struct rpi_firmware {
+ 	struct kref consumers;
  };
  
 +static struct platform_device *g_pdev;
@@ -36,32 +36,33 @@ Signed-off-by: Noralf Trønnes <noralf@tronnes.org>
  static DEFINE_MUTEX(transaction_lock);
  
  static void response_callback(struct mbox_client *cl, void *msg)
-@@ -249,6 +251,7 @@ static int rpi_firmware_probe(struct pla
- 	init_completion(&fw->c);
+@@ -272,6 +274,7 @@ static int rpi_firmware_probe(struct pla
+ 	kref_init(&fw->consumers);
  
  	platform_set_drvdata(pdev, fw);
 +	g_pdev = pdev;
  
  	rpi_firmware_print_firmware_revision(fw);
  	rpi_register_hwmon_driver(dev, fw);
-@@ -276,6 +279,7 @@ static int rpi_firmware_remove(struct pl
- 	platform_device_unregister(rpi_clk);
- 	rpi_clk = NULL;
- 	mbox_free_channel(fw->chan);
-+	g_pdev = NULL;
+@@ -301,6 +304,8 @@ static int rpi_firmware_remove(struct pl
+ 
+ 	rpi_firmware_put(fw);
  
++	g_pdev = NULL;
++
  	return 0;
  }
-@@ -288,7 +292,7 @@ static int rpi_firmware_remove(struct pl
+ 
+@@ -314,7 +319,7 @@ static int rpi_firmware_remove(struct pl
   */
  struct rpi_firmware *rpi_firmware_get(struct device_node *firmware_node)
  {
 -	struct platform_device *pdev = of_find_device_by_node(firmware_node);
 +	struct platform_device *pdev = g_pdev;
+ 	struct rpi_firmware *fw;
  
  	if (!pdev)
- 		return NULL;
-@@ -312,7 +316,18 @@ static struct platform_driver rpi_firmwa
+@@ -352,7 +357,18 @@ static struct platform_driver rpi_firmwa
  	.shutdown	= rpi_firmware_shutdown,
  	.remove		= rpi_firmware_remove,
  };
diff --git a/target/linux/bcm27xx/patches-5.10/950-0073-raspberrypi-firmware-Export-the-general-transaction-.patch b/target/linux/bcm27xx/patches-5.10/950-0073-raspberrypi-firmware-Export-the-general-transaction-.patch
index 8450d5f48ab1..17f36af72d99 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0073-raspberrypi-firmware-Export-the-general-transaction-.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0073-raspberrypi-firmware-Export-the-general-transaction-.patch
@@ -13,7 +13,7 @@ Signed-off-by: Eric Anholt <eric@anholt.net>
 
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -43,7 +43,7 @@ static void response_callback(struct mbo
+@@ -46,7 +46,7 @@ static void response_callback(struct mbo
   * Sends a request to the firmware through the BCM2835 mailbox driver,
   * and synchronously waits for the reply.
   */
@@ -22,7 +22,7 @@ Signed-off-by: Eric Anholt <eric@anholt.net>
  rpi_firmware_transaction(struct rpi_firmware *fw, u32 chan, u32 data)
  {
  	u32 message = MBOX_MSG(chan, data);
-@@ -68,6 +68,7 @@ rpi_firmware_transaction(struct rpi_firm
+@@ -71,6 +71,7 @@ rpi_firmware_transaction(struct rpi_firm
  
  	return ret;
  }
diff --git a/target/linux/bcm27xx/patches-5.10/950-0079-firmware-raspberrypi-Notify-firmware-of-a-reboot.patch b/target/linux/bcm27xx/patches-5.10/950-0079-firmware-raspberrypi-Notify-firmware-of-a-reboot.patch
index bdcf5c4b42bf..cedbabdc749f 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0079-firmware-raspberrypi-Notify-firmware-of-a-reboot.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0079-firmware-raspberrypi-Notify-firmware-of-a-reboot.patch
@@ -13,7 +13,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -11,6 +11,7 @@
+@@ -12,6 +12,7 @@
  #include <linux/module.h>
  #include <linux/of_platform.h>
  #include <linux/platform_device.h>
@@ -21,7 +21,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
  #include <linux/slab.h>
  #include <soc/bcm2835/raspberrypi-firmware.h>
  
-@@ -177,6 +178,26 @@ int rpi_firmware_property(struct rpi_fir
+@@ -180,6 +181,26 @@ int rpi_firmware_property(struct rpi_fir
  }
  EXPORT_SYMBOL_GPL(rpi_firmware_property);
  
@@ -48,7 +48,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
  static void
  rpi_firmware_print_firmware_revision(struct rpi_firmware *fw)
  {
-@@ -318,15 +339,32 @@ static struct platform_driver rpi_firmwa
+@@ -359,15 +380,32 @@ static struct platform_driver rpi_firmwa
  	.remove		= rpi_firmware_remove,
  };
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch b/target/linux/bcm27xx/patches-5.10/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch
index ac331f34dc93..9149c7e791a2 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0087-firmware-raspberrypi-Add-backward-compatible-get_thr.patch
@@ -16,15 +16,15 @@ Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>
 
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -28,6 +28,7 @@ struct rpi_firmware {
+@@ -29,6 +29,7 @@ struct rpi_firmware {
  	struct mbox_chan *chan; /* The property channel. */
  	struct completion c;
  	u32 enabled;
 +	u32 get_throttled;
- };
  
- static struct platform_device *g_pdev;
-@@ -174,6 +175,12 @@ int rpi_firmware_property(struct rpi_fir
+ 	struct kref consumers;
+ };
+@@ -177,6 +178,12 @@ int rpi_firmware_property(struct rpi_fir
  
  	kfree(data);
  
@@ -37,7 +37,7 @@ Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>
  	return ret;
  }
  EXPORT_SYMBOL_GPL(rpi_firmware_property);
-@@ -198,6 +205,27 @@ static int rpi_firmware_notify_reboot(st
+@@ -201,6 +208,27 @@ static int rpi_firmware_notify_reboot(st
  	return 0;
  }
  
@@ -65,7 +65,7 @@ Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>
  static void
  rpi_firmware_print_firmware_revision(struct rpi_firmware *fw)
  {
-@@ -227,6 +255,11 @@ rpi_register_hwmon_driver(struct device
+@@ -230,6 +258,11 @@ rpi_register_hwmon_driver(struct device
  
  	rpi_hwmon = platform_device_register_data(dev, "raspberrypi-hwmon",
  						  -1, NULL, 0);
diff --git a/target/linux/bcm27xx/patches-5.10/950-0093-firmware-raspberrypi-Report-the-fw-variant-during-pr.patch b/target/linux/bcm27xx/patches-5.10/950-0093-firmware-raspberrypi-Report-the-fw-variant-during-pr.patch
index 64df0f7c7f76..6cfca9d4f2e5 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0093-firmware-raspberrypi-Report-the-fw-variant-during-pr.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0093-firmware-raspberrypi-Report-the-fw-variant-during-pr.patch
@@ -27,7 +27,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
 
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -231,6 +231,15 @@ rpi_firmware_print_firmware_revision(str
+@@ -234,6 +234,15 @@ rpi_firmware_print_firmware_revision(str
  {
  	time64_t date_and_time;
  	u32 packet;
@@ -43,7 +43,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
  	int ret = rpi_firmware_property(fw,
  					RPI_FIRMWARE_GET_FIRMWARE_REVISION,
  					&packet, sizeof(packet));
-@@ -240,7 +249,35 @@ rpi_firmware_print_firmware_revision(str
+@@ -243,7 +252,35 @@ rpi_firmware_print_firmware_revision(str
  
  	/* This is not compatible with y2038 */
  	date_and_time = packet;
@@ -80,7 +80,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
  }
  
  static void
-@@ -309,6 +346,7 @@ static int rpi_firmware_probe(struct pla
+@@ -332,6 +369,7 @@ static int rpi_firmware_probe(struct pla
  	g_pdev = pdev;
  
  	rpi_firmware_print_firmware_revision(fw);
diff --git a/target/linux/bcm27xx/patches-5.10/950-0183-Initialise-rpi-firmware-before-clk-bcm2835.patch b/target/linux/bcm27xx/patches-5.10/950-0183-Initialise-rpi-firmware-before-clk-bcm2835.patch
index dcd9ff43463b..4235df82f06a 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0183-Initialise-rpi-firmware-before-clk-bcm2835.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0183-Initialise-rpi-firmware-before-clk-bcm2835.patch
@@ -36,7 +36,7 @@ Co-authored-by: Phil Elwell <phil@raspberrypi.org>
  MODULE_DESCRIPTION("BCM2835 clock driver");
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -430,7 +430,7 @@ out2:
+@@ -471,7 +471,7 @@ out2:
  out1:
  	return ret;
  }
diff --git a/target/linux/bcm27xx/patches-5.10/950-0201-mmc-sdhci-Silence-MMC-warnings.patch b/target/linux/bcm27xx/patches-5.10/950-0201-mmc-sdhci-Silence-MMC-warnings.patch
index 071aa0287c6a..e015cc81c63f 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0201-mmc-sdhci-Silence-MMC-warnings.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0201-mmc-sdhci-Silence-MMC-warnings.patch
@@ -22,7 +22,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
  
  #define MAX_TUNING_LOOP 40
  
-@@ -3126,7 +3126,7 @@ static void sdhci_timeout_timer(struct t
+@@ -3127,7 +3127,7 @@ static void sdhci_timeout_timer(struct t
  	spin_lock_irqsave(&host->lock, flags);
  
  	if (host->cmd && !sdhci_data_line_cmd(host->cmd)) {
@@ -31,7 +31,7 @@ Signed-off-by: Maxime Ripard <maxime@cerno.tech>
  		       mmc_hostname(host->mmc));
  		sdhci_dumpregs(host);
  
-@@ -3148,7 +3148,7 @@ static void sdhci_timeout_data_timer(str
+@@ -3149,7 +3149,7 @@ static void sdhci_timeout_data_timer(str
  
  	if (host->data || host->data_cmd ||
  	    (host->cmd && sdhci_data_line_cmd(host->cmd))) {
diff --git a/target/linux/bcm27xx/patches-5.10/950-0370-firmware-raspberrypi-Add-support-for-tryonce-reboot-.patch b/target/linux/bcm27xx/patches-5.10/950-0370-firmware-raspberrypi-Add-support-for-tryonce-reboot-.patch
index ea48f5ac89d2..2fa1b068e7fb 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0370-firmware-raspberrypi-Add-support-for-tryonce-reboot-.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0370-firmware-raspberrypi-Add-support-for-tryonce-reboot-.patch
@@ -20,7 +20,7 @@ mechanism to be implemented for OS upgrades.
 
 --- a/drivers/firmware/raspberrypi.c
 +++ b/drivers/firmware/raspberrypi.c
-@@ -191,6 +191,7 @@ static int rpi_firmware_notify_reboot(st
+@@ -194,6 +194,7 @@ static int rpi_firmware_notify_reboot(st
  {
  	struct rpi_firmware *fw;
  	struct platform_device *pdev = g_pdev;
@@ -28,7 +28,7 @@ mechanism to be implemented for OS upgrades.
  
  	if (!pdev)
  		return 0;
-@@ -199,8 +200,28 @@ static int rpi_firmware_notify_reboot(st
+@@ -202,8 +203,28 @@ static int rpi_firmware_notify_reboot(st
  	if (!fw)
  		return 0;
  
diff --git a/target/linux/bcm27xx/patches-5.10/950-0408-drm-atomic-Pass-the-full-state-to-CRTC-atomic-enable.patch b/target/linux/bcm27xx/patches-5.10/950-0408-drm-atomic-Pass-the-full-state-to-CRTC-atomic-enable.patch
index 3953a1d0ab8a..fd16a2bd4df8 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0408-drm-atomic-Pass-the-full-state-to-CRTC-atomic-enable.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0408-drm-atomic-Pass-the-full-state-to-CRTC-atomic-enable.patch
@@ -698,7 +698,7 @@ Link: https://patchwork.freedesktop.org/patch/msgid/845aa10ef171fc0ea060495efef1
  	struct mdp5_crtc_state *mdp5_cstate = to_mdp5_crtc_state(crtc->state);
 --- a/drivers/gpu/drm/mxsfb/mxsfb_kms.c
 +++ b/drivers/gpu/drm/mxsfb/mxsfb_kms.c
-@@ -303,7 +303,7 @@ static void mxsfb_crtc_atomic_flush(stru
+@@ -343,7 +343,7 @@ static void mxsfb_crtc_atomic_flush(stru
  }
  
  static void mxsfb_crtc_atomic_enable(struct drm_crtc *crtc,
@@ -707,7 +707,7 @@ Link: https://patchwork.freedesktop.org/patch/msgid/845aa10ef171fc0ea060495efef1
  {
  	struct mxsfb_drm_private *mxsfb = to_mxsfb_drm_private(crtc->dev);
  	struct drm_device *drm = mxsfb->drm;
-@@ -327,7 +327,7 @@ static void mxsfb_crtc_atomic_enable(str
+@@ -367,7 +367,7 @@ static void mxsfb_crtc_atomic_enable(str
  }
  
  static void mxsfb_crtc_atomic_disable(struct drm_crtc *crtc,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0409-drm-atomic-Pass-the-full-state-to-CRTC-atomic_check.patch b/target/linux/bcm27xx/patches-5.10/950-0409-drm-atomic-Pass-the-full-state-to-CRTC-atomic_check.patch
index 86481e8955c4..b123b9da1bd7 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0409-drm-atomic-Pass-the-full-state-to-CRTC-atomic_check.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0409-drm-atomic-Pass-the-full-state-to-CRTC-atomic_check.patch
@@ -638,7 +638,7 @@ Link: https://patchwork.freedesktop.org/patch/msgid/20201028123222.1732139-1-max
  	 * and that we don't have conflicting mixer stages:
 --- a/drivers/gpu/drm/mxsfb/mxsfb_kms.c
 +++ b/drivers/gpu/drm/mxsfb/mxsfb_kms.c
-@@ -270,17 +270,19 @@ static void mxsfb_crtc_mode_set_nofb(str
+@@ -310,17 +310,19 @@ static void mxsfb_crtc_mode_set_nofb(str
  }
  
  static int mxsfb_crtc_atomic_check(struct drm_crtc *crtc,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0410-drm-atomic-Pass-the-full-state-to-CRTC-atomic-begin-.patch b/target/linux/bcm27xx/patches-5.10/950-0410-drm-atomic-Pass-the-full-state-to-CRTC-atomic-begin-.patch
index 86a918a40e19..9696a45390af 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0410-drm-atomic-Pass-the-full-state-to-CRTC-atomic-begin-.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0410-drm-atomic-Pass-the-full-state-to-CRTC-atomic-begin-.patch
@@ -680,7 +680,7 @@ Link: https://patchwork.freedesktop.org/patch/msgid/20201028123222.1732139-2-max
  	struct mdp5_crtc_state *mdp5_cstate = to_mdp5_crtc_state(crtc->state);
 --- a/drivers/gpu/drm/mxsfb/mxsfb_kms.c
 +++ b/drivers/gpu/drm/mxsfb/mxsfb_kms.c
-@@ -286,7 +286,7 @@ static int mxsfb_crtc_atomic_check(struc
+@@ -326,7 +326,7 @@ static int mxsfb_crtc_atomic_check(struc
  }
  
  static void mxsfb_crtc_atomic_flush(struct drm_crtc *crtc,
diff --git a/target/linux/bcm27xx/patches-5.10/950-0686-drm-Use-the-state-pointer-directly-in-atomic_check.patch b/target/linux/bcm27xx/patches-5.10/950-0686-drm-Use-the-state-pointer-directly-in-atomic_check.patch
index 2d16e5a2f722..2b3c03379e32 100644
--- a/target/linux/bcm27xx/patches-5.10/950-0686-drm-Use-the-state-pointer-directly-in-atomic_check.patch
+++ b/target/linux/bcm27xx/patches-5.10/950-0686-drm-Use-the-state-pointer-directly-in-atomic_check.patch
@@ -75,7 +75,7 @@ Link: https://patchwork.freedesktop.org/patch/msgid/20201102133834.1176740-3-max
  static void drm_simple_kms_crtc_enable(struct drm_crtc *crtc,
 --- a/drivers/gpu/drm/mxsfb/mxsfb_kms.c
 +++ b/drivers/gpu/drm/mxsfb/mxsfb_kms.c
-@@ -282,7 +282,7 @@ static int mxsfb_crtc_atomic_check(struc
+@@ -322,7 +322,7 @@ static int mxsfb_crtc_atomic_check(struc
  		return -EINVAL;
  
  	/* TODO: Is this needed ? */
diff --git a/target/linux/bcm53xx/patches-5.10/905-BCM53573-minor-hacks.patch b/target/linux/bcm53xx/patches-5.10/905-BCM53573-minor-hacks.patch
index a1d27949f048..9aaa72595d0c 100644
--- a/target/linux/bcm53xx/patches-5.10/905-BCM53573-minor-hacks.patch
+++ b/target/linux/bcm53xx/patches-5.10/905-BCM53573-minor-hacks.patch
@@ -21,7 +21,7 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
  	clocks {
 --- a/drivers/bcma/main.c
 +++ b/drivers/bcma/main.c
-@@ -330,14 +330,6 @@ static int bcma_register_devices(struct
+@@ -328,14 +328,6 @@ static int bcma_register_devices(struct
  	}
  #endif
  
@@ -36,7 +36,7 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
  #ifdef CONFIG_BCMA_NFLASH
  	if (bus->drv_cc.nflash.present) {
  		err = platform_device_register(&bcma_nflash_dev);
-@@ -415,6 +407,14 @@ int bcma_bus_register(struct bcma_bus *b
+@@ -413,6 +405,14 @@ int bcma_bus_register(struct bcma_bus *b
  			bcma_register_core(bus, core);
  	}
  
diff --git a/target/linux/generic/hack-5.10/410-block-fit-partition-parser.patch b/target/linux/generic/hack-5.10/410-block-fit-partition-parser.patch
index 95067125fb3c..c417c78eb6c2 100644
--- a/target/linux/generic/hack-5.10/410-block-fit-partition-parser.patch
+++ b/target/linux/generic/hack-5.10/410-block-fit-partition-parser.patch
@@ -1,6 +1,6 @@
 --- a/block/blk.h
 +++ b/block/blk.h
-@@ -353,6 +353,7 @@ char *disk_name(struct gendisk *hd, int
+@@ -355,6 +355,7 @@ char *disk_name(struct gendisk *hd, int
  #define ADDPART_FLAG_NONE	0
  #define ADDPART_FLAG_RAID	1
  #define ADDPART_FLAG_WHOLEDISK	2
diff --git a/target/linux/generic/hack-5.10/904-debloat_dma_buf.patch b/target/linux/generic/hack-5.10/904-debloat_dma_buf.patch
index a447e6b81334..cd63206d4321 100644
--- a/target/linux/generic/hack-5.10/904-debloat_dma_buf.patch
+++ b/target/linux/generic/hack-5.10/904-debloat_dma_buf.patch
@@ -72,7 +72,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 +MODULE_LICENSE("GPL");
 --- a/kernel/sched/core.c
 +++ b/kernel/sched/core.c
-@@ -3049,6 +3049,7 @@ int wake_up_state(struct task_struct *p,
+@@ -3062,6 +3062,7 @@ int wake_up_state(struct task_struct *p,
  {
  	return try_to_wake_up(p, state, 0);
  }
diff --git a/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch b/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
index 4abdce85d017..ebff78a19264 100644
--- a/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
+++ b/target/linux/generic/pending-5.10/670-ipv6-allow-rejecting-with-source-address-failed-policy.patch
@@ -157,7 +157,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	case RTN_THROW:
  	case RTN_UNREACHABLE:
  	default:
-@@ -4426,6 +4445,17 @@ static int ip6_pkt_prohibit_out(struct n
+@@ -4429,6 +4448,17 @@ static int ip6_pkt_prohibit_out(struct n
  	return ip6_pkt_drop(skb, ICMPV6_ADM_PROHIBITED, IPSTATS_MIB_OUTNOROUTES);
  }
  
@@ -175,7 +175,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  /*
   *	Allocate a dst for local (unicast / anycast) address.
   */
-@@ -4906,7 +4936,8 @@ static int rtm_to_fib6_config(struct sk_
+@@ -4909,7 +4939,8 @@ static int rtm_to_fib6_config(struct sk_
  	if (rtm->rtm_type == RTN_UNREACHABLE ||
  	    rtm->rtm_type == RTN_BLACKHOLE ||
  	    rtm->rtm_type == RTN_PROHIBIT ||
@@ -185,7 +185,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  		cfg->fc_flags |= RTF_REJECT;
  
  	if (rtm->rtm_type == RTN_LOCAL)
-@@ -6076,6 +6107,8 @@ static int ip6_route_dev_notify(struct n
+@@ -6079,6 +6110,8 @@ static int ip6_route_dev_notify(struct n
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  		net->ipv6.ip6_prohibit_entry->dst.dev = dev;
  		net->ipv6.ip6_prohibit_entry->rt6i_idev = in6_dev_get(dev);
@@ -194,7 +194,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  		net->ipv6.ip6_blk_hole_entry->dst.dev = dev;
  		net->ipv6.ip6_blk_hole_entry->rt6i_idev = in6_dev_get(dev);
  #endif
-@@ -6087,6 +6120,7 @@ static int ip6_route_dev_notify(struct n
+@@ -6090,6 +6123,7 @@ static int ip6_route_dev_notify(struct n
  		in6_dev_put_clear(&net->ipv6.ip6_null_entry->rt6i_idev);
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  		in6_dev_put_clear(&net->ipv6.ip6_prohibit_entry->rt6i_idev);
@@ -202,7 +202,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  		in6_dev_put_clear(&net->ipv6.ip6_blk_hole_entry->rt6i_idev);
  #endif
  	}
-@@ -6278,6 +6312,8 @@ static int __net_init ip6_route_net_init
+@@ -6281,6 +6315,8 @@ static int __net_init ip6_route_net_init
  
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  	net->ipv6.fib6_has_custom_rules = false;
@@ -211,7 +211,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	net->ipv6.ip6_prohibit_entry = kmemdup(&ip6_prohibit_entry_template,
  					       sizeof(*net->ipv6.ip6_prohibit_entry),
  					       GFP_KERNEL);
-@@ -6288,11 +6324,21 @@ static int __net_init ip6_route_net_init
+@@ -6291,11 +6327,21 @@ static int __net_init ip6_route_net_init
  			 ip6_template_metrics, true);
  	INIT_LIST_HEAD(&net->ipv6.ip6_prohibit_entry->rt6i_uncached);
  
@@ -234,7 +234,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	net->ipv6.ip6_blk_hole_entry->dst.ops = &net->ipv6.ip6_dst_ops;
  	dst_init_metrics(&net->ipv6.ip6_blk_hole_entry->dst,
  			 ip6_template_metrics, true);
-@@ -6319,6 +6365,8 @@ out:
+@@ -6322,6 +6368,8 @@ out:
  	return ret;
  
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
@@ -243,7 +243,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  out_ip6_prohibit_entry:
  	kfree(net->ipv6.ip6_prohibit_entry);
  out_ip6_null_entry:
-@@ -6338,6 +6386,7 @@ static void __net_exit ip6_route_net_exi
+@@ -6341,6 +6389,7 @@ static void __net_exit ip6_route_net_exi
  	kfree(net->ipv6.ip6_null_entry);
  #ifdef CONFIG_IPV6_MULTIPLE_TABLES
  	kfree(net->ipv6.ip6_prohibit_entry);
@@ -251,7 +251,7 @@ Signed-off-by: Jonas Gorski <jogo@openwrt.org>
  	kfree(net->ipv6.ip6_blk_hole_entry);
  #endif
  	dst_entries_destroy(&net->ipv6.ip6_dst_ops);
-@@ -6415,6 +6464,9 @@ void __init ip6_route_init_special_entri
+@@ -6418,6 +6467,9 @@ void __init ip6_route_init_special_entri
  	init_net.ipv6.ip6_prohibit_entry->rt6i_idev = in6_dev_get(init_net.loopback_dev);
  	init_net.ipv6.ip6_blk_hole_entry->dst.dev = init_net.loopback_dev;
  	init_net.ipv6.ip6_blk_hole_entry->rt6i_idev = in6_dev_get(init_net.loopback_dev);
diff --git a/target/linux/generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch b/target/linux/generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch
index cb02c7182963..511a9f755513 100644
--- a/target/linux/generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch
+++ b/target/linux/generic/pending-5.10/800-bcma-get-SoC-device-struct-copy-its-DMA-params-to-th.patch
@@ -50,20 +50,14 @@ Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
  	if (err)
 --- a/drivers/bcma/main.c
 +++ b/drivers/bcma/main.c
-@@ -236,12 +236,16 @@ EXPORT_SYMBOL(bcma_core_irq);
- 
- void bcma_prepare_core(struct bcma_bus *bus, struct bcma_device *core)
- {
-+	struct device *dev = &core->dev;
-+
- 	core->dev.release = bcma_release_core_dev;
+@@ -241,8 +241,10 @@ void bcma_prepare_core(struct bcma_bus *
  	core->dev.bus = &bcma_bus_type;
  	dev_set_name(&core->dev, "bcma%d:%d", bus->num, core->core_index);
  	core->dev.parent = bus->dev;
 -	if (bus->dev)
 +	if (bus->dev) {
  		bcma_of_fill_device(bus->dev, core);
-+		dma_coerce_mask_and_coherent(dev, bus->dev->coherent_dma_mask);
++		dma_coerce_mask_and_coherent(&core->dev, bus->dev->coherent_dma_mask);
 +	}
  
  	switch (bus->hosttype) {
diff --git a/target/linux/octeontx/patches-5.10/0004-PCI-add-quirk-for-Gateworks-PLX-PEX860x-switch-with-.patch b/target/linux/octeontx/patches-5.10/0004-PCI-add-quirk-for-Gateworks-PLX-PEX860x-switch-with-.patch
index e33a28e9539b..507ce638253f 100644
--- a/target/linux/octeontx/patches-5.10/0004-PCI-add-quirk-for-Gateworks-PLX-PEX860x-switch-with-.patch
+++ b/target/linux/octeontx/patches-5.10/0004-PCI-add-quirk-for-Gateworks-PLX-PEX860x-switch-with-.patch
@@ -21,8 +21,8 @@ Signed-off-by: Tim Harvey <tharvey@gateworks.com>
 +#include <linux/of.h>
  #include <linux/platform_data/x86/apple.h>
  #include <linux/pm_runtime.h>
- #include <linux/switchtec.h>
-@@ -5625,3 +5626,34 @@ static void apex_pci_fixup_class(struct
+ #include <linux/suspend.h>
+@@ -5707,3 +5708,34 @@ static void apex_pci_fixup_class(struct
  }
  DECLARE_PCI_FIXUP_CLASS_HEADER(0x1ac1, 0x089a,
  			       PCI_CLASS_NOT_DEFINED, 8, apex_pci_fixup_class);
